Runtimes/Supplemental/Synchronization/CMakeLists.txt

```
@@ -0,0 +1,106 @@
+
+set(${PROJECT_NAME}_VENDOR_MODULE_DIR "${CMAKE_SOURCE_DIR}/../cmake/modules/vendor"
+  CACHE FILEPATH "Location for private build system extension")
+
+find_package(SwiftCore QUIET REQUIRED)
+
+include(gyb)
+include(ResourceEmbedding)
+include(AvailabilityMacros)
+include(PlatformInfo)
+
+include("${${PROJECT_NAME}_VENDOR_MODULE_DIR}/Settings.cmake" OPTIONAL)
+
+option(${PROJECT_NAME}_ENABLE_LIBRARY_EVOLUTION "Generate ABI resilient runtime libraries"
+  ${SwiftCore_ENABLE_LIBRARY_EVOLUTION})
+
+option(${PROJECT_NAME}_ENABLE_PRESPECIALIZATION "Enable generic metadata prespecialization"
+  ${SwiftCore_ENABLE_PRESPECIALIZATION})
+
+include(CatalystSupport)
+include(EmitSwiftInterface)
+
+add_compile_options(
+  "$<$<COMPILE_LANGUAGE:Swift>:-nostdlibimport>"
+  "$<$<AND:$<BOOL:${${PROJECT_NAME}_ENABLE_LIBRARY_EVOLUTION}>,$<COMPILE_LANGUAGE:Swift>>:-enable-library-evolution>"
+  "$<$<AND:$<BOOL:${${PROJECT_NAME}_ENABLE_PRESPECIALIZATION}>,$<COMPILE_LANGUAGE:Swift>>:SHELL:-Xfrontend -prespecialize-generic-metadata>"
+  $<$<COMPILE_LANGUAGE:Swift>:-explicit-module-build>
+  $<$<COMPILE_LANGUAGE:Swift>:-enable-builtin-module>
+  $<$<COMPILE_LANGUAGE:Swift>:-strict-memory-safety>
+  "$<$<COMPILE_LANGUAGE:Swift>:SHELL:-enable-experimental-feature RawLayout>"
+  "$<$<COMPILE_LANGUAGE:Swift>:SHELL:-enable-experimental-feature StaticExclusiveOnly>"
+  "$<$<COMPILE_LANGUAGE:Swift>:SHELL:-enable-experimental-feature Extern>")
+
+gyb_expand(Atomics/AtomicIntegers.swift.gyb Atomics/AtomicIntegers.swift)
+gyb_expand(Atomics/AtomicStorage.swift.gyb Atomics/AtomicStorage.swift)
+
+add_library(swiftSynchronization
+  Atomics/Atomic.swift
+  Atomics/AtomicBool.swift
+  Atomics/AtomicFloats.swift
+  Atomics/AtomicLazyReference.swift
+  Atomics/AtomicMemoryOrderings.swift
+  Atomics/AtomicOptional.swift
+  Atomics/AtomicPointers.swift
+  Atomics/AtomicRepresentable.swift
+  Atomics/WordPair.swift
+  Atomics/AtomicStorage.swift
+  Atomics/AtomicIntegers.swift
+  Cell.swift
+  Mutex/Mutex.swift
+  $<$<PLATFORM_ID:Darwin>:Mutex/DarwinImpl.swift>
+  $<$<PLATFORM_ID:Linux>:Mutex/LinuxImpl.swift>
+  $<$<PLATFORM_ID:WASI>:Mutex/SpinLoopHint.swift>
+  $<$<PLATFORM_ID:WINDOWS>:Mutex/WindowsImpl.swift>)
+
+set_target_properties(swiftSynchronization PROPERTIES
+  Swift_MODULE_NAME Synchronization)
+
+target_link_libraries(swiftSynchronization
+  PRIVATE
+    swiftCore
+    $<$<PLATFORM_ID:Darwin>:swiftDarwin>)
+
+set(${PROJECT_NAME}_INSTALL_LIBDIR "${CMAKE_INSTALL_LIBDIR}/swift$<$<NOT:$<BOOL:${BUILD_SHARED_LIBS}>>:_static>$<$<BOOL:${Supplemental_INSTALL_NESTED_SUBDIR}>:/${Supplemental_PLATFORM_SUBDIR}/${Supplemental_ARCH_SUBDIR}>" CACHE STRING "")
+set(${PROJECT_NAME}_INSTALL_SWIFTMODULEDIR "${CMAKE_INSTALL_LIBDIR}/swift$<$<NOT:$<BOOL:${BUILD_SHARED_LIBS}>>:_static>$<$<BOOL:${Supplemental_INSTALL_NESTED_SUBDIR}>:/${Supplemental_PLATFORM_SUBDIR}>" CACHE STRING "")
+
+install(TARGETS swiftSynchronization
+  EXPORT SwiftSupplementalTargets
```

Probably it does not make sense to reference `Supplemental` here (since each library can be build standalone)
```suggestion
  EXPORT SwiftSynchronizationTargets
```